main.tf

```diff
@@ -101,6 +101,11 @@ resource "google_compute_instance" "instances" {
 }
 
 resource "null_resource" "instance-prereq" {
+  # Changes to any instance of the cluster requires re-provisioning
+  triggers {
+    current_ec2_instance_id = "${element(google_compute_instance.instances.*.id, count.index)}"
```

Comment:
its not ec2. This var should be renamed

Reply:
Updated the variable name and made it simpler to keep it consistent with all the cloud providers.